CryptoBinPkg/Driver/readme.md

```diff
@@ -1,54 +1,99 @@
-# Crypto Driver
+# Shared Crypto Binaries
 
-VERSION: 2.0
+Shared crypto consists of a set of pre-built binaries that provide cryptographic services. The services are exposed
+to other modules via dynamic interfaces such as PPIs in the PEI phase and protocols in the DXE and MM phases
 
-This is a pre-built version of BaseCryptLib and TlsLib built into binaries that make the services available to other
-modules via dynamic interfaces such as PPIs and protocols.
+**This project does not directly implement crypto algorithms**. Crypto algorithm implementation is provided by an
+underlying crypto provider. The provided may change over time based on industry shifts in crypto requirements,
+enhancements for embedded crypto support in available crypto projects, and overall crypto provider quality and
+maintainership.
 
-This currently supports Openssl implementation of BaseCryptLib but this will be expanded in the future.
+- Currently, the OpenSSL crypto project is used.
 
-## Benefits
+Any consumer of this binary should expect that the underlying crypto provider may change. Consumers are encouraged to
+participate in all aspects of shared crypto including code contributions and feedback.
+
+## Terminology
 
-But first, why would you care about this?
+- **Crypto Provider** - The project that provides an implementation of cryptographic algorithms used by this project
+  those crypto services to platform firmware.
+  - Examples: [OpenSSL](https://github.com/openssl/openssl), [Mbed TLS](https://github.com/Mbed-TLS/mbedtls),
+    [SymCrypt](https://github.com/microsoft/SymCrypt).
+- **Flavor** - A collection of cryptographic algorithms included in a given shared crypto binary.
+- **Phase** - The "boot phase" that the binary applies to where a phase is defined in the [Platform Initialization
+  Specification](https://uefi.org/specs/PI/1.8A/).
+  - Examples: `PEIM`, `DXE_DRIVER`, `DXE_SMM_DRIVER`, etc.
+- **Platform Firmware** - A firmware project that integrates the shared crypto binaries made available by this project.
+  An open-source example of platform firmware is the [Mu Tiano Platforms firmware repo](https://github.com/microsoft/mu_tiano_platforms).
+- **Shared Crypto** - The name of this project. Shared crypto means a set of pre-built binaries that provide
+  cryptographic services.

```

Comment:
Maybe adding something like:

```suggestion
  cryptographic services via protocols, ensuring that all modules share a crypto implementation, instead of each module having their own implementation linked in.

```

I think that one of the confusing parts of this is why there are these binaries, but then every module still links in a version of BaseCryptLib to do the protocol calling.

Reply:
Thanks for pointing that out. Its one of those things we might take for granted after talking about this for so long. I added more elaboration.